SimMuon/Configuration/python/SimMuon_cff.py

```diff
@@ -29,3 +29,9 @@
 run3_GEM.toReplaceWith( muonDigi, _run3_muonDigi )
 from Configuration.Eras.Modifier_phase2_muon_cff import phase2_muon
 phase2_muon.toReplaceWith( muonDigi, _phase2_muonDigi )
+
+from SimMuon.MCTruth.muonSimClassificationByHits_cff import *
```

Comment:
why does this need to be imported in this file?
SimMuon_cff is supposed to be used for digitization, not as a one-stop shop that can import everything from SimMuon/ subsystem

Reply:
muonSimClassifier and related are now showing up in the DIGI step config file,
which may be somewhat safe, but is also apparently unnecessary.

Reply:
I really don't understand what's wrong. Do you want me to put it SimMuonBlah_cff instead SimMuon_cff?

Reply:
``` SimMuon.MCTruth.muonSimClassificationByHits_cff import``` can be removed from this file

Reply:
Step: DIGIPREMIX_S2 Spec:
Traceback (most recent call last):
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/bin/slc6_amd64_gcc630/cmsDriver.py", line 55, in <module>
    run()
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/bin/slc6_amd64_gcc630/cmsDriver.py", line 27, in run
    configBuilder.prepare()
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/Applications/ConfigBuilder.py", line 2090, in prepare
    self.addStandardSequences()
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/Applications/ConfigBuilder.py", line 744, in addStandardSequences
    getattr(self,"prepare_"+stepName)(sequence = getattr(self,stepName+"DefaultSeq"))
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/Applications/ConfigBuilder.py", line 1445, in prepare_DIGIPREMIX_S2
    self.loadDefaultOrSpecifiedCFF(sequence,self.DIGIDefaultCFF)
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/Applications/ConfigBuilder.py", line 1191, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(defaultCFF)
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/Applications/ConfigBuilder.py", line 321, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_9_4_X_2017-10-23-1200/python/FWCore/ParameterSet/Config.py", line 582, in load
    module = __import__(moduleName)
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/StandardSequences/DigiDMPreMix_cff.py", line 24, in <module>
    from SimMuon.Configuration.SimMuon_cff import *
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/SimMuon/Configuration/SimMuon_cff.py", line 34, in <module>
    muonSimClassificationByHitsTask
NameError: name 'muonSimClassificationByHitsTask' is not defined

Reply:
I hope that error is readable. I simply removed the line.

Reply:
yes, it's readable, the simMuonTask should go as well.

Reply:
I really have no clue where it's all going. 
  File "/data/dmytro/projects/CMSSW_9_4_X_2017-10-23-1200/python/Configuration/StandardSequences/RecoSim_cff.py", line 5, in <module>
    recosim = cms.Task( simMuonTask )
NameError: name 'simMuonTask' is not defined

Do you want simply to put "from SimMuon.MCTruth.muonSimClassificationByHits_cff import *" into Configuration/StandardSequences/python/RecoSim_cff.py and change "recosim = cms.Task( simMuonTask )" to "recosim = cms.Task( muonSimClassificationByHitsTask )"?

Reply:
yes, please decouple SimMuon.Configuration.SimMuon_cff (which should not depend on reco inputs) from  SimMuon.MCTruth. which can